### DISCUSSION TITLE
Enable to opt out of pagefind

### DISCUSSION THREAD

**Discussion by justin5267** [withastro/starlight#952] [Feature Requests]
### What version of `starlight` are you using?

0.5.2

### What is your idea?

Add support for disabling pagefind componet globally

### Why is this feature necessary?

when using third-party search services, pagefind still generates indexes during project deployment, which can be time-consuming.

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by delucis** [comment]
Thanks for the suggestion!

Search customisation is something we want to add, for example for people who would like to add an alternative service instead of the default Pagefind solution, so this is probably something we should tackle together with that.

**Comment by justin5267** [comment]
Thanks for your reply and could you help me with how to disable Pagefind from generating indexes? I've looked around but don't know how to turn it off.

<img width="524" alt="image" src="https://github.com/withastro/starlight/assets/105055890/e065a0ad-d0c3-4fe1-a7b6-3ca872a550ee">

**Reply by stereobooster** [reply]
it should be easy to add config here https://github.com/withastro/starlight/blob/8c0b267fd99fa21c9bf16319e087db4c0a5e14fd/packages/starlight/index.ts#L79-L90

**Reply by justin5267** [reply]
Many thanks!

**Comment by delucis** [comment]
It’s not currently possible. We’d need to add a configuration option for this.

**Comment by delucis** [comment]
This will be possible in our next release — v0.14 🎉 

You will be able to set a new `pagefind` option in the Starlight integration config:

```js
starlight({
  pagefind: false,
})
```

Docs: https://starlight.astro.build/reference/configuration/#pagefind
